chain/src/main/scala/org/bitcoins/chain/models/BlockHeaderDAO.scala

```diff
@@ -47,6 +47,32 @@ sealed abstract class BlockHeaderDAO
     table.filter(_.hash.inSet(hashes))
   }
 
+  /** Retrives the ancestor for the given block header at the given height
+    * @param child
+    * @param height
+    * @return
+    */
+  def getAncestorAtHeight(
+      child: BlockHeaderDb,
+      height: Long): Future[Option[BlockHeaderDb]] = {
+    //extremely inefficient and probably needs to be re-implemented
+    //we are reading from disk every time here
+    val headerOptF = read(child.previousBlockHashBE)
```

Comment:
probably want to replace this with a single read that takes in perhaps a little too much (i.e. filter by height)